dockerfile-image-update/src/main/java/com/salesforce/dockerfileimageupdate/utils/GithubAppCheck.java

```diff
@@ -0,0 +1,103 @@
+import org.bouncycastle.util.io.pem.PemReader;
+import org.kohsuke.github.GitHub;
+import org.kohsuke.github.GitHubBuilder;
+import org.kohsuke.github.HttpException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.File;
+import java.io.FileReader;
+import java.io.IOException;
+import java.security.GeneralSecurityException;
+import java.security.KeyFactory;
+import java.security.Security;
+import java.security.interfaces.RSAPrivateKey;
+import java.security.spec.PKCS8EncodedKeySpec;
+import java.time.Instant;
+import java.util.Date;
+
+public class GithubAppCheck {
+    private static final Logger log = LoggerFactory.getLogger(GithubAppCheck.class);
+
+    private final String appId;
+    private final String privateKeyPath;
+    private String jwt;
+    private Instant jwtExpiry;
+    private GitHub gitHub;
+
+    public GithubAppCheck(final Namespace ns){
+        this.appId = ns.get(Constants.SKIP_GITHUB_APP_ID);
+        this.privateKeyPath = ns.get(Constants.SKIP_GITHUB_APP_KEY);
+        this.jwt = null;
+        this.jwtExpiry = null;
+        try {
+            generateJWT(this.appId, this.privateKeyPath);
+        } catch (GeneralSecurityException | IOException exception) {
+            log.warn("Could not initialise JWT due to exception: {}", exception.getMessage());
+        }
+        try {
+            this.gitHub = new GitHubBuilder()
+                    .withEndpoint(CommandLine.gitApiUrl(ns))
+                    .withJwtToken(jwt)
+                    .build();
+        } catch (IOException exception) {
+            log.warn("Could not initialise github due to exception: {}", exception.getMessage());
+            this.gitHub = null;
+        }
+    }
+
+    protected boolean isGithubAppEnabledOnRepository(String fullRepoName){
+        refreshJwtIfNeeded(appId, privateKeyPath);
+        try {
+            gitHub.getApp().getInstallationByRepository(fullRepoName.split("/")[0], fullRepoName.split("/")[1]);
+            return true;
+        } catch (HttpException exception) {
+            if (exception.getResponseCode() != 404) {
+                log.warn("Caught a HTTPException {} while trying to get app installation. Defaulting to False", exception.getMessage());
+            }
+            return false;
+        } catch (IOException exception) {
+            log.warn("Caught a IOException {} while trying to get app installation. Defaulting to False", exception.getMessage());
+            return false;
+        }
+    }
+
+    private void refreshJwtIfNeeded(String appId, String privateKeyPath){
+        if (jwt == null || jwtExpiry.isBefore(Instant.now().minusSeconds(60))) {  // Adding a buffer to ensure token validity
+            try {
+                generateJWT(appId, privateKeyPath);
+            } catch (IOException | GeneralSecurityException exception) {
+                log.warn("Could not refresh the JWT due to exception: {}", exception.getMessage());
```

Comment:
Wouldn't this cause a problem? If we simply ignore this exception and default the value to false in the `isGithubAppEnabledOnRepository` method, it means that if the JWT refresh fails, we'll end up creating PRs even if the repository isn't onboarded to Renovate.

Reply:
Same comment as above. I am trying to be cautious. Lmk if you think otherwise.

Reply:
@jeetchoudhary Why would we end up creating PRs even if the repository isn't onboarded to Renovate (if they are not onboarded to Renovate then Renovate should not create a PR no matter if the JWT token fails or not right)? You mean even if the repository that is not even onboarded to DFIU?

Reply:
> Yes, but I am trying to be cautious. Since we are dealing with docker patches, isn't it better to get both DFIU and Renovate PRs and have a confusing experience, rather than not receive either PR and fall behind on your patches?

I agree with Sai in this case